graylog2-server/src/main/java/org/graylog/plugins/formatting/units/provider/SupportedUnitsProvider.java

```diff
@@ -0,0 +1,97 @@
+ * Server Side Public License for more details.
+ *
+ * You should have received a copy of the Server Side Public License
+ * along with this program. If not, see
+ * <http://www.mongodb.com/licensing/server-side-public-license>.
+ */
+package org.graylog.plugins.formatting.units.provider;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import jakarta.inject.Inject;
+import jakarta.inject.Provider;
+import jakarta.inject.Singleton;
+import jakarta.validation.constraints.NotNull;
+import org.graylog.plugins.formatting.units.model.BaseUnit;
+import org.graylog.plugins.formatting.units.model.Conversion;
+import org.graylog.plugins.formatting.units.model.DerivedUnit;
+import org.graylog.plugins.formatting.units.model.SupportedUnits;
+import org.graylog.plugins.formatting.units.model.Unit;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+
+import static org.graylog.plugins.formatting.units.model.Conversion.ConversionAction.DIVIDE;
+import static org.graylog.plugins.formatting.units.model.Conversion.ConversionAction.MULTIPLY;
+
+
+@Singleton
+public class SupportedUnitsProvider implements Provider<SupportedUnits> {
+
+    private static final List<Unit> UNITS_FALLBACK = List.of(
+            new BaseUnit("B", "byte", "size"),
+            new DerivedUnit("kB", "kilobyte", "size", new Conversion(1000, MULTIPLY)),
+            new DerivedUnit("MB", "megabyte", "size", new Conversion(1000 * 1000, MULTIPLY)),
+            new DerivedUnit("GB", "gigabyte", "size", new Conversion(1000 * 1000 * 1000, MULTIPLY)),
+
+            new DerivedUnit("ns", "nanosecond", "time", new Conversion(1000 * 1000 * 1000, DIVIDE)),
+            new DerivedUnit("μs", "microsecond", "time", new Conversion(1000 * 1000, DIVIDE)),
+            new DerivedUnit("ms", "millisecond", "time", new Conversion(1000, DIVIDE)),
+            new BaseUnit("s", "second", "time"),
+            new DerivedUnit("min", "minute", "time", new Conversion(60, MULTIPLY)),
+            new DerivedUnit("h", "hour", "time", new Conversion(60 * 60, MULTIPLY)),
+            new DerivedUnit("d", "day", "time", new Conversion(24 * 60 * 60, MULTIPLY)),
+            new DerivedUnit("m", "month", "time", new Conversion(30 * 24 * 60 * 60, MULTIPLY)),
+
+            new BaseUnit("%", "percent", "percent")
+    );
+    private static final String SUPPORTED_UNITS_RES_FILE_LOCATION = "/units/supported_units.json";
+    private SupportedUnits supportedUnitsMemoized;
+
+    private final ObjectMapper objectMapper;
+
+    @Inject
+    public SupportedUnitsProvider(final ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public SupportedUnits get() {
+        if (this.supportedUnitsMemoized == null) {
+            try {
+                this.supportedUnitsMemoized = loadFromFile(objectMapper);
+            } catch (Exception ex) {
+                this.supportedUnitsMemoized = useFallback();
```

Comment:
What do we need the fallback for? If we cannot read the bundled resource file, then this is a programmatic error and should be raised as loud as possible. Silently falling back to a shorter list of units would just cover this up.

Reply:
I wanted to be sure that basic units used in HardcodedFieldUnitObtainingMethod are loaded even if loading from file fails. I can remove it if you don't like it.